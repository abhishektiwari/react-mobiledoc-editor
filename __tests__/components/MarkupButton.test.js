import React from 'react';
import { shallow } from 'enzyme';
import MarkupButton from '../../src/components/MarkupButton';

describe('<MarkupButton />', () => {
  it('should render', () => {
    const wrapper = shallow(<MarkupButton tag="A" type="reset" title="Link" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const wrapper = shallow(<MarkupButton tag="A">Link</MarkupButton>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle markup on click', () => {
    const editor = { toggleMarkup: jest.fn() };
    const context = { editor };
    const wrapper = shallow(<MarkupButton tag='A' />, { context });
    wrapper.find('button').simulate('click');
    expect(editor.toggleMarkup).toBeCalledWith('A');
  });

  it('should set active class', () => {
    const context = { activeMarkupTags: ['a']};
    const wrapper = shallow(<MarkupButton tag='A' className="keep" />, { context });
    expect(wrapper).toMatchSnapshot();
  });
});
