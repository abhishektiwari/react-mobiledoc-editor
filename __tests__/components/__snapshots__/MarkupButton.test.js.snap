// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarkupButton /> should render 1`] = `
<button
  className=""
  onClick={[Function]}
  title="Link"
  type="reset"
>
  A
</button>
`;

exports[`<MarkupButton /> should render children 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  Link
</button>
`;

exports[`<MarkupButton /> should set active class 1`] = `
<button
  className="keep active"
  onClick={[Function]}
  type="button"
>
  A
</button>
`;
